### ISSUE TITLE
CompositeTVSimulator batch Processing(losing hair)

### ISSUE THREAD

**Issue by tinyjapaneselunches** [tsulej/GenerateMe#17]
Very new and accidentally left this in another issue folder. Can someone explain batch processing to me. I have everything set up the right way but don't get the process. If I'm just processing a folder, do I need the filename filled out? I did that and get a NullPointerException error. Also, if I do get this right...you just press 'b'? Do you press the play button first? Well I would think yes, obviously, but I'm just trying to make sense of all this.

**Comment by genmeblog** [comment]
Generally all of this scripts require some preparation, especially batch processing (sorry, no convenient gui :/ ).
Can I assume you can run a sketch and produce the result for a single file?
If answer is yes, do the following:

1. create a subfolder, say `images`
2. put all your images into this folder
3. be sure [this](https://github.com/tsulej/GenerateMe/blob/master/composite_video_simulator/composite_video_simulator.pde#L45) is set to your file extension `.jpg` for jpegs, `.png` for pngs.
4. replace [this](https://github.com/tsulej/GenerateMe/blob/master/composite_video_simulator/composite_video_simulator.pde#L46) line with `String foldername = "./images";`
5. run the script
6. press `b` and wait

**Comment by genmeblog** [comment]
One more thing came to my mind: set [the filename](https://github.com/tsulej/GenerateMe/blob/master/composite_video_simulator/composite_video_simulator.pde#L44) to the exising file.

**Comment by tinyjapaneselunches** [comment]
Sorry...just saw this. I can't run the script. This is what it says:
NullPointerException
"The file "./imagesimages.jpg" is missing or inaccessible, make sure the URL is valid or that the file has been added to your sketch and is readable." All the settings you told me are the ones I just did and that's the error message I get. Also, when I run the script and get the error, it leads me down the code to a certain point and highlights this:
buffer.beginDraw();

Any ideas? Again....ultra Noob/Padawan here.

**Comment by tinyjapaneselunches** [comment]
Just tried it again and now it's highlighting this:
buffer = createGraphics(img.width, img.height);

**Comment by tinyjapaneselunches** [comment]
Nope. Still the same message. It again brings me down to line 109 which is:
buffer = createGraphics(img.width, img.height);

I think this line of code is what needs to get changed maybe? Not sure, but the error message always leads me to this particlular line.

**Comment by tinyjapaneselunches** [comment]
Can anyone answer this question for me? Kinda time sensitive.

**Comment by genmeblog** [comment]
Which version of Processing you are using? Should be 2.x
Can you run the version from github?
Do you have composite_video_simulator.pde and test.jpg files inside "composite_video_simulator/" folder?

**Comment by tinyjapaneselunches** [comment]
I'm running 2.21. That was the recommended version from when I first downloaded the generateme scripts. I'll try running on the github version and get back to you ASAP. Everything is in their respective folders. I've been using these scripts for quite a while for artwork and this is the first time I'm trying batch processing. I'll let you know if the github version works in a minute.

**Comment by genmeblog** [comment]
ok, ready to support you

**Comment by genmeblog** [comment]
Ok. works on my side:

Folder structure is like that:

![image](https://user-images.githubusercontent.com/38646601/224803800-b6992458-6e50-450f-bfcc-637a77338632.png)

Setup is like following:

```
// set up filename
String filename = "test1";
String fileext = ".jpg";
String foldername = "./images/";
```

Press 'b'

```
[...]
test5.jpg... to YIQ
composite chroma lowpass
chroma to luma
preemphasis
video noise
chroma from luma
fm
chroma noise
chroma phase noise
vhs luma lowpass
vhs chroma lowpass
chroma vert blend
vhs sharpen
svideo out
chroma loss
composite recombination
composite chroma lowpass out
from YIQ
done
render scanlines
saved
results saved in ./images/A34630B7 folder
```

![image](https://user-images.githubusercontent.com/38646601/224804755-a2f36509-d75a-455c-9572-d0674a656fb8.png)


Please note that there should exist file "./images/test1.jpg".

**Comment by tinyjapaneselunches** [comment]
I'll try this. Should they all be labeled as 'test' for the images?

**Comment by tinyjapaneselunches** [comment]
I'll take that as a yes.

**Comment by tinyjapaneselunches** [comment]
SUCCESS!!!!! That worked!! Thank you sooooo much! Now all I need to do is grow my hair back. This is awesome!!!

**Comment by genmeblog** [comment]
Great! Share the final result (if possible)!

**Comment by tinyjapaneselunches** [comment]
I'm working on 2 computers but I'll transfer the files and upload the results.

**Comment by Apsynth00** [comment]
I had a problem with
-**NullPointerException**: My solution is check the format File for example if file Test is "Test.jpeg" You need put :
String filename = "Test"; // should have 4:3 ratio
String fileext = "**.jpeg**";
String foldername = "./";

//If the file is "test.jpg" 

String filename = "Test"; // should have 4:3 ratio
String fileext = "**.jpg**";
String foldername = "./";

The photos almost always has a .Jpeg Format.

**Comment by genmeblog** [comment]
I'm pretty sure it should work on both `.jpg` and `.jpeg` extensions. Processing accepts both (this is the same format btw). I would verify file names (some filesystems are case sensitive some are not).
